### PR TITLE
feat: Containerize the envless platform.

### PR DESCRIPTION
**Pull request by Aju100**
# Description

## Type of change

> Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

```
docker build -t name-of-container .
```

> make sure you have installed docker on your system.

**Test Configuration**:

- OS with version:
- Browser with version:
- SDK / CLI with version:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I documented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Aju100** [comment]
@dahal @chetannn can u try this on for testing one.
```sh
clone the repo
docker build -t envless .
docker run envless -p 3000:3000
```

Also try this one via docker-compose.yml
```sh
docker-compose up
```

We have an issue as dotenv is not found, may I know how can I install this as dotenv is unavailable that leads to failure of running turbo command. You can see by above command. I tried using ```npm install dotenv```.
> make sure docker and docker-compose is installed out there.

**Comment by Aju100** [comment]
Things to do
- Partition container image to run separate for runner runtime. 
- Fixed single bundle container image for platform core and resolve every dependencies issue
- Need to resolve docker-compose issue and connection with redis, postgresql
- So on as pipeline for container one later on.

**Comment by dahal** [comment]
> 

Its this CLI we are using
https://github.com/entropitor/dotenv-cli

Also, I tried with this CLI
image


But I am getting turbo error
image

**Comment by Aju100** [comment]
@dahal any issue with this ?
```
docker build -t envless .
docker run envless -p 3000:3000
```

**Comment by Aju100** [comment]
Its working from my side on my machine for minor POC let me know though we have an issue on our codebase
```sh
platform    | envless:dev: error - ./pages/_app.tsx:7:0
platform    | envless:dev: Module not found: Can't resolve 'ui/styles/tailwind.css'
platform    | envless:dev:    5 | import { SessionProvider } from "next-auth/react";
platform    | envless:dev:    6 | import NextNProgress from "nextjs-progressbar";
platform    | envless:dev: >  7 | import "ui/styles/tailwind.css";
platform    | envless:dev:    8 | import ErrorBoundary from "@/components/errorboundary";
platform    | envless:dev:    9 | import Fallback from "@/components/errorboundary/Fallback";
platform    | envless:dev:   10 | import "@/styles/balloon.css";
```
cc @dahal @chetannn

**Comment by dahal** [comment]
Yea, I ran into same issue as well.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  platform:
+    container_name: platform
+    build:
+      context: .
+      dockerfile: Dockerfile
+    restart: always
+    env_file: .env.example
+    ports:
+      - 3000:3000
```
Comment (dahal) [inline docker-compose.yml:45]:
```suggestion
      - 3000:3000
      - 3883:3883
```

We also need to expose port `3883:3883` (currently used by email interceptor)

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - envless_network
+    depends_on:
+      - postgres
+      - redis-rest
+
+networks:
+  envless_network:
+
+volumes:
+  postgres_data:
```
Comment (dahal) [inline docker-compose.yml:56]:
```suggestion
  postgres_data:

```

File: docker-compose.yml (hunk 1 above)
Comment (Aju100) [reply docker-compose.yml:45]: Got it.

File: apps/platform/.gitignore [hunk 3]
```diff
@@ -1,3 +1,3 @@
 .mailing
 node_modules
-.env
+.env
```
Comment (swapnil-talpade) [inline apps/platform/.gitignore:3]:
```suggestion
.env

```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# Build the project
+COPY --from=builder /app/out/full/ .
+
+ENV SKIP_ENV_VALIDATION true
+ENV EMAIL_FROM envless@example.com
+RUN npx turbo run db:generate
+
+RUN yarn turbo run build --filter=platform...
+
+CMD ["yarn","dev"]
```
Comment (swapnil-talpade) [inline Dockerfile:38]:
```suggestion
CMD ["yarn","dev"]

```

**Comment by Aju100** [comment]
@dahal @chetannn closing this PR for the newly Fresh PR one.

**Pull request closed without merging**